src/libraries/Native/Unix/Common/pal_compiler.h

```diff
@@ -19,4 +19,14 @@
 #define c_static_assert(e) c_static_assert_msg(e, "")
 #endif
 
+#ifndef DLLEXPORT
 #define DLLEXPORT __attribute__ ((__visibility__ ("default")))
```

Comment:
It may be a good idea to explicitly strip exporting of the shim entrypoints when compiling them as part of CoreCLR. They do not need to be exported.

Maybe define a new macro `PALEXPORT` and change all shims to use it. The macro can then be defined as needed depending on the context.

Reply:
Sounds good.

Reply:
It seems like I can't do this yet for S.G.Native because if so then mono will not find the entrypoints as they're still PInvoking into S.G.Native. We will be able to do this until we link mono into it.

Reply:
This looks fine. You may be able to do it by adding:
 ```
// Suppress exporting of the PAL APIs
add_definitions(-DPALEXPORT)
```
to `libraries-native/CMakeLists.txt`.

Reply:
I see. Hmm, that's a good idea. Let me try that.